src/Settings/TerminalSettings.vala

```diff
@@ -46,9 +47,60 @@ namespace ElementaryTweaks {
             return instance;
         }
 
+        public static Gtk.ListStore get_cursor_shapes (out int active_index) {
+            var shape_layouts = new Gtk.ListStore(2, typeof (string), typeof (string));
+            Gtk.TreeIter iter;
+
+            int index = 0;
+            active_index = 0;
+
+            var shapes_map = new Gee.HashMap<string, string> ();
+            shapes_map.set ("Block", _("Block"));
+            shapes_map.set ("I-Beam", _("I-Beam"));
+            shapes_map.set ("Underline", _("Underline"));
+
+            foreach (var layout in shapes_map.entries) {
+                shape_layouts.append (out iter);
+                shape_layouts.set (iter, 0, layout.value, 1, layout.key);
+                if (TerminalSettings.get_default ().tab_bar_behavior == layout.key) {
+                    active_index = index;
+                }
+
+                index++;
+            }
+
+            return shape_layouts;
+        }
+
+        public static Gtk.ListStore get_tab_behaviours (out int active_index) {
+            var behaviour_layouts = new Gtk.ListStore(2, typeof (string), typeof (string));
+            Gtk.TreeIter iter;
+
+            int index = 0;
+            active_index = 0;
+
+            var behaviour_map = new Gee.HashMap<string, string> ();
+            behaviour_map.set ("Always Show Tabs", _("Always Show Tabs"));
+            behaviour_map.set ("Hide When Single Tab", _("Hide When Single Tab"));
+            behaviour_map.set ("Never Show Tabs", _("Never Show Tabs"));
+
+            foreach (var layout in behaviour_map.entries) {
+                behaviour_layouts.append (out iter);
+                behaviour_layouts.set (iter, 0, layout.value, 1, layout.key);
+                if (TerminalSettings.get_default ().cursor_shape == layout.key) {
+                    active_index = index;
+                }
+
+                index++;
+            }
+
+            return behaviour_layouts;
+        }
+
         public void reset () {
             string[] to_reset = {"background", "cursor-color", "font", "foreground", "palette",
                                  "scrollback-lines", "unsafe-paste-alert", "natural-copy-paste",
+                                 "tab-bar-behaviour",
```

Comment:
```suggestion
                                 "tab-bar-behavior",
```

You seem to misspell the value name of GSettings. This causes clicking the reset button doesn't restore the value.